CI: linuxPrivilegedIPv4AndIPv6TestSuite/TestNodeChurnXFRMLeaks

## CI failure

Hit on https://github.com/cilium/cilium/pull/27242: https://github.com/cilium/cilium/actions/runs/5752260375/job/15592941475

```
--- FAIL: Test (227.53s)
    --- FAIL: Test/linuxPrivilegedIPv4AndIPv6TestSuite (3.51s)
        --- FAIL: Test/linuxPrivilegedIPv4AndIPv6TestSuite/TestNodeChurnXFRMLeaks (1.06s)
            node_linux_test.go:809: 
                ... value *errors.joinError = &errors.joinError{errs:[]error{(*errors.joinError)(0xc0010e0180)}} ("failed to enable ipsec with out IPv6 using local IP ::/0, rem 2001:aaaa::/64, spi 0: unable to replace remote state: file exists")
```

--- a/pkg/datapath/linux/node_linux_test.go
+++ b/pkg/datapath/linux/node_linux_test.go
@@ -752,7 +752,6 @@ func (s *linuxPrivilegedBaseTestSuite) TestNodeUpdateIDs(c *check.C) {
 
 // Tests that we don't leak XFRM policies and states as nodes come and go.
 func (s *linuxPrivilegedBaseTestSuite) TestNodeChurnXFRMLeaks(c *check.C) {
-	externalNodeDevice := "ipsec_interface"
 
 	// Cover the XFRM configuration for IPAM modes cluster-pool, kubernetes, etc.
 	config := datapath.LocalNodeConfiguration{
@@ -760,7 +759,20 @@ func (s *linuxPrivilegedBaseTestSuite) TestNodeChurnXFRMLeaks(c *check.C) {
 		EnableIPv6:  s.enableIPv6,
 		EnableIPSec: true,
 	}
-	//s.testNodeChurnXFRMLeaksWithConfig(c, config)
+	s.testNodeChurnXFRMLeaksWithConfig(c, config)
+}
+
+// Tests the same as linuxPrivilegedBaseTestSuite.TestNodeChurnXFRMLeaks just
+// for the subnet encryption. IPv4-only because of https://github.com/cilium/cilium/issues/27280.
+func (s *linuxPrivilegedIPv4OnlyTestSuite) TestNodeChurnXFRMLeaks(c *check.C) {
+	externalNodeDevice := "ipsec_interface"
+
+	// Cover the XFRM configuration for IPAM modes cluster-pool, kubernetes, etc.
+	config := datapath.LocalNodeConfiguration{
+		EnableIPv4:  s.enableIPv4,
+		EnableIPSec: true,
+	}
+	s.testNodeChurnXFRMLeaksWithConfig(c, config)
 
 	// In the case of subnet encryption (tested below), the IPsec logic
 	// retrieves the IP address of the encryption interface directly so we need

